consumer/src/main/kotlin/streams/kafka/KafkaEventSink.kt

```diff
@@ -4,109 +4,133 @@ import kotlinx.coroutines.Dispatchers
-        this.db = db
+    private val mappingKeys = mapOf("timeout" to "streams.sink.polling.interval",
+            "from" to "kafka.auto.offset.reset")
+
+    override fun getEventConsumerFactory(): StreamsEventConsumerFactory {
+        return object: StreamsEventConsumerFactory() {
+            override fun createStreamsEventConsumer(config: Map<String, String>, log: Log): StreamsEventConsumer<*> {
+                val kafkaConfig = KafkaSinkConfiguration.from(config)
+                val kafkaConsumer = KafkaConsumer<String, ByteArray>(kafkaConfig.asProperties())
+                return KafkaEventConsumer(kafkaConsumer, kafkaConfig.streamsSinkConfiguration, log)
+            }
+        }
     }
 
     override fun start() {
-        this.streamsTopicService = StreamsTopicService(this.db, kafkaConfig.streamsSinkConfiguration)
-        if (streamsTopicService!!.getTopics().isEmpty()) {
-            log.info("No topic configuration found under streams.sink.topic.*, Kafka Sink will not stared")
+        val streamsConfig = StreamsSinkConfiguration.from(config)
+        if (!streamsConfig.enabled) {
+            log.info("The sink will not started, please set the property streams.sink.enabled=true")
             return
         }
+        log.info("Starting the Kafka Sink")
+        val topics = streamsTopicService.getAll()
+        this.eventConsumer = getEventConsumerFactory()
+                .createStreamsEventConsumer(config.raw, log)
+                .withTopics(topics.keys)
+        this.eventConsumer.start()
+        if (log.isDebugEnabled) {
+            log.debug("Subscribed topics with queries: $topics")
+        }
+        this.job = createJob()
+        log.info("Kafka Sink started")
+    }
 
-        this.queryExecution = StreamsEventSinkQueryExecution(this.streamsTopicService!!, db, log)
-        createConsumer()
-
-        job = createJob()
-        log.info("Kafka Sink Connector started.")
+    override fun stop() {
+        log.info("Stopping Sink daemon Job")
+        this.eventConsumer.stop()
+        StreamsUtils.ignoreExceptions({ job.cancel() }, UninitializedPropertyAccessException::class.java)
     }
 
-    private fun createConsumer() {
-        this.kafkaConsumer = KafkaConsumer(kafkaConfig.asProperties())
-        kafkaConsumer.subscribe(streamsTopicService!!.getTopics())
-        if (log.isDebugEnabled) {
-            log.debug("Subscribing topics: ${streamsTopicService!!.getAll()}")
-        } else {
-            log.info("Subscribing topics: ${streamsTopicService!!.getTopics()}")
+    override fun getEventSinkConfigMapper(): StreamsEventSinkConfigMapper {
+        return object: StreamsEventSinkConfigMapper(streamsConfigMap, mappingKeys) {
+            override fun convert(config: Map<String, String>): Map<String, String> {
+                val props = streamsConfigMap
+                        .toMutableMap()
+                props.putAll(config.mapKeys { mappingKeys.getOrDefault(it.key, it.key) })
```

Comment:
what does this do again?
btw. youc an just do `= streamsConfigMap + config.mapKeys { mappingKeys.getOrDefault(it.key, it.key) }`
no need for the mutable map.

Reply:
Ok!

Reply:
It merges the default configuration such as zookeeper and broker addresses provided in the Neo4j conf with the custom that comes from the `{config}` param of the procedure